src/server.rs

```diff
@@ -0,0 +1,213 @@
+use crate::onion;
+use crate::secp::{self, Commitment, ComSignature, SecretKey};
+use crate::ser;
+use crate::types::Onion;
+
+use jsonrpc_derive::rpc;
+use jsonrpc_http_server::*;
+use jsonrpc_http_server::jsonrpc_core::*;
+use jsonrpc_core::{Result, Value};
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use std::sync::Mutex;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct ServerConfig {
+    pub key: SecretKey,
+    pub addr: SocketAddr,
+    pub is_first: bool,

```

Comment:
It should be configured to true for node 1, false for node 2...n
It's not clear yet what ServerConfig will look like in its final form, so that field is likely to change.

Reply:
I think it might be worthwhile to know that you are server i out of n.
That allows you to check for the correct size of payloads, and whether you're the final node.

Reply:
Good suggestion. Milestone 3 is where ServerConfig will really come into play. I will make note of this for when I get to milestone 3.